### DISCUSSION TITLE
Directory or file recover with Drat

### DISCUSSION THREAD

**Discussion by designt** [jivanpal/apfs-tools#17] [Q&A]
Hello,

After I delete "iCloud Dive (archive)" on my Mac OS 10.15.4 (APFS partition) I lose some important files.
Is it possible to use Drat to recover trashed files or directory ?
Would you pls give me some tips to do it ?
Thanks a lot.

**Comment by jivanpal** [comment]
Unfortunately undelete is not currently possible, as we haven't yet implemented the ability to explore past APFS checkpoints. Full recover/undelete support is planned for 0.2.1 or 0.2.2. The ETA for that is realistically around 6 months (perhaps more, perhaps less) as recent personal issues have delayed my projects/work and may put Drat development on hold for a while.
